### ISSUE TITLE
import error with pytest / tests directory import

### ISSUE THREAD

**Issue by wyardley** [PyCQA/pylint#3594]
### Steps to reproduce
1. We have a layout like this:
```
├── __init__.py
├── bin
│   └── pollinate
├── lib
│   ├── __init__.py
[...]
│   └── writers
│       ├── __init__.py
│       ├── file
│       │   ├── __init__.py
│       │   ├── docker.py
│       │   └── kubernetes.py
│       └── template
│           ├── __init__.py
│           └── env_factory.py
└── tests
    ├── __init__.py
    [...]
    ├── fixtures
    │   └── metadata.json
    └── test_make_configs.py
```
We recognize that this may be a slightly non-standard layout, but it's worked Ok in the past.
`tests/test_make_configs.py` has:
```python
from lib import (get_package_name,
                 get_package_version,
                 Deployment)

from lib.writers.file import kubernetes
```

Before, we also had a:
```python
sys.path.append(os.path.dirname(sys.path[0]))   # noqa: E402
```
but switched to using `python3 -m pytest` based on the suggestions in https://docs.pytest.org/en/latest/goodpractices.html

2. Run pylint >= 2.5.0 on the codebase (2.4.x seems to work Ok)

### Current behavior
```
tests/test_make_configs.py:23: [E0401(import-error), ] Unable to import 'lib'
tests/test_make_configs.py:27: [E0401(import-error), ] Unable to import 'lib.writers.file'
```

### Expected behavior
```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```
Open to suggestions on how to improve the setup / imports if there's a way that will work better with pylint. That said, the two different styles of import pass under 2.4.x. The resulting error is similar to the ones around running in parallel mode, only 2.5.2 still errors as well.

### pylint --version output
```
pylint 2.5.2
astroid 2.4.1
Python 3.7.6 (default, Jan 21 2020, 20:27:32) 
[Clang 11.0.0 (clang-1100.0.33.16)]
```

**Comment by thisch** [comment]
In which file did you have the 

```
sys.path.append(os.path.dirname(sys.path[0]))   # noqa: E402
```
?

Is the `__init__.py` file in the following tree a top-level `__init__.py`?
```
├── __init__.py
├── bin
....
```
I don't understand why you need this top-level `__init__.py`, since according to the description `lib`, `bin`, ... seem to be top-level python-pkgs.

**Comment by wyardley** [comment]
@thatch thanks! I think it may just be vestigial. I hadn't noticed it until I went to put these details in, and I can confirm that removing it does resolve the pylint failure.

**Comment by wyardley** [comment]
Solve on our end; feel free to close, though FWIW, the older version does, rightly or not, tolerate this a bit better.
